app.go

```diff
@@ -26,14 +26,27 @@ func app(cmd *cobra.Command, args []string) {
 	var err error
 
 	// Get version of Docker benchmark to run
-	if dockerVersion != "" {
-		version = dockerVersion
-	} else {
-		version, err = getDockerVersion()
-		if err != nil {
+	if benchmarkVersion != "" {
+		if dockerVersion != "" {
 			util.ExitWithError(
-				fmt.Errorf("Version check failed: %s\nAlternatively, you can specify the version with --version",
-					err))
+				fmt.Errorf("Version check failed: Error can't specify both --version and --benchmark flags\nIf not specify neither docker version will be auto detect"))
+		}
+		version = benchmarkVersion
+	} else {
+		if dockerVersion != "" {
+			version, err = getDockerCisVersion(dockerVersion)
+			if err != nil {
+				util.ExitWithError(
+					fmt.Errorf("Version check failed: %s\nAlternatively, you can specify the version with --version",
+						err))
+			}
+		} else {
+			version, err = getDockerVersion()
+			if err != nil {
+				util.ExitWithError(
+					fmt.Errorf("Version check failed: %s\nAlternatively, you can specify the version with --version",
+						err))
+			}
```

Comment:
What you have will work but I think it could be neater. Whether the user suggests a Docker version (with `--version`), or whether it gets auto-detected, we need to convert that into a CIS benchmark with the `getDockerCisVersion()` function. I would call that function just once, from here, and not also call it from `getDockerVersion()`. So the logic would be like this:   

```
dockerVersion is the user-supplied docker version
if dockerVersion is empty:
  dockerVersion = auto-detected dockerVersion
benchmarkVersion = convert dockerVersion to benchmark version
```

The changed code would be something like this: 
```suggestion
		if dockerVersion == "" {
		        // Auto-detect the version of Docker that's running
			dockerVersion, err = getDockerVersion()
			if err != nil {
				util.ExitWithError(
					fmt.Errorf("Version check failed: %s\nAlternatively, you can specify the version with --version",
						err))
		}
		
		// Find the CIS benchmark version for this version of Docker
		version, err = getDockerCisVersion(dockerVersion)
		if err != nil {
			util.ExitWithError(
				fmt.Errorf("Failed to get a valid CIS benchmark version for Docker version %s: %v",
					dockerVersion, err))
		}
```

Note that you'd have to remove the call to `getDockerCisVersion()` at 139. But also note that as it currently stands, the comment for `getDockerVersion()` at L132 is incorrect - it says it returns the docker server engine version but if you leave it calling `getDockerCisVersion()` then it's really returning the CIS benchmark version.